### ISSUE TITLE
Adaoter for Victron VRM portal

### ISSUE THREAD

**Issue by gidhap** [ioBroker/AdapterRequests#374]
Although several registers of the Victron PV inverters are accessible via Modbus, it is difficult to calculate energy consumption and  generation. Victron provides a nice portal ([the VRM portal ](https://vrm.victronenergy.com/installation-overview)) where all relevant data and statistics are available. 
I'd love to be able to include this data into my ioBroker setup.
Good thing is... there is an[ API provided by Victron.](https://docs.victronenergy.com/vrmapi/overview.html#introduction)
even as a [PDF](https://docs.victronenergy.com/vrmapi/overview.pdf)
Unfortunately I have no clue about coding...

Hope to get some thumbs up for this one

**Comment by gidhap** [comment]
@Apollon77 I could give you access to my VRM portal, so you don't need hardware, hope this helps :)

**Comment by Apollon77** [comment]
Thank you and the developer that finds this issue would be happy to also get that offen ;-) It most likely will not be me because I have too many other duties on my plate :-)

**Comment by gidhap** [comment]
worth a try  ;o)

**Comment by mbad82** [comment]
Also interested!

**Comment by weigejer** [comment]
would also be interested. greetings Jörg

**Comment by elkja** [comment]
would also be interested

**Comment by SimonBard** [comment]
I think I might be able to help out here. Can someone mail me login-data for the account? 
mail-user is notmyham, I am using gmail.

**Comment by gidhap** [comment]
@SimonBard I sent you an e-mail :)

**Comment by rbagia** [comment]
any update here my friends?

**Comment by gidhap** [comment]
no news so far on my side, else than I replaced my Victron stuff recently.

**Comment by blaumsass** [comment]
i´ve been interrestet

**Comment by Apollon77** [comment]
For all "me too" poster ... if you do not vote in the initial post here with a thumbs up NOONE will see it ... and please do that and no "Me too" posts. Thank you

**Comment by derAlff** [comment]
For information: At the moment I develop a adapter for Victron Energy devices 😎

**Comment by mcm1957** [comment]
@derAlff 
Thanks for the info and especially for your effort and spending your time.
Please post the link to the repository as soon as you have committed some usaeable code.

**Comment by derAlff** [comment]
Hey ho, no problem. I need my Energy Sorage in my ioBroker... so I can spend the code 👌

At the moment I'm in the testing phase. Is this solved, I commit the code to GitHub and post the link here and in the forum too 👍

**Comment by derAlff** [comment]
Dear all,

I have created an adapter. Now, can you tell me your preferred values you want?

For test respectively at the moment I read the following values:
1. Mode (Grid, Shore,...)
2. Consumption on L1, L2, L3 (voltage, amperes an watts)
3. Grid usage on L1, L2, L3 (voltage, amperes an watts)
4. Inverter production on L1, L2, L3 (voltage, amperes an watts)

**Comment by derAlff** [comment]
Hey ho, here is the Link to the repo.
Please test it ;)

https://github.com/derAlff/ioBroker.ve

**Comment by Apollon77** [comment]
Is "ve" as name really the best choice?

**Comment by derAlff** [comment]
No, not the best. But I am talking with Victron Energy for usage of name and logo the last days. If that is not clear I use "ve" as name 😅

**Comment by HolgerCR** [comment]
to set some values over Iobroker whpuld be fine , example i use also the Charger so i can set the cahrcing to 8 instead 16... 
To read the voltage about the batteries whould also be fine and many more values thanks in advance

**Comment by incase** [comment]
@derAlff IIUC, you are using the modbus interface. The request here was for using the VRM API though. The VRM API has one very large plus over the modbus interface: It allows reading forecast data.

curl --request GET   --url "https://vrmapi.victronenergy.com/v2/installations/<VRM_portal_ID>/stats?type=custom&attributeCodes[]=vrm_pv_inverter_yield_fc&interval=hours&start=<unix_timestamp_now>&end=<unix_timestamp_later>"   --header 'Content-Type: application/json'   --header 'x-authorization: Bearer <token_from_login>'

where VRM_portal_ID is the token number (usually 6 digits, I guess) that is also part of the URL when you are logged in via a browser) and <token_from_login> is what you get when you send a login request. The request here asks for the predicted inverter yield (if you have a PV inverter connected to the GX device via modbus / sunspec), vrm_pv_charger_yield_fc would return the predicted yield from an MPPT.

**Comment by derAlff** [comment]
> @derAlff IIUC, you are using the modbus interface. The request here was for using the VRM API though. The VRM API has one very large plus over the modbus interface: It allows reading forecast data.
> 
> 
> 
> curl --request GET   --url "https://vrmapi.victronenergy.com/v2/installations/<VRM_portal_ID>/stats?type=custom&attributeCodes[]=vrm_pv_inverter_yield_fc&interval=hours&start=<unix_timestamp_now>&end=<unix_timestamp_later>"   --header 'Content-Type: application/json'   --header 'x-authorization: Bearer <token_from_login>'
> 
> 
> 
> where VRM_portal_ID is the token number (usually 6 digits, I guess) that is also part of the URL when you are logged in via a browser) and <token_from_login> is what you get when you send a login request. The request here asks for the predicted inverter yield (if you have a PV inverter connected to the GX device via modbus / sunspec), vrm_pv_charger_yield_fc would return the predicted yield from an MPPT.

Parallel I create an adapter for using VRM. But this adapter is not ready for tests. Please wait for release.

**Comment by KIRK-NRW** [comment]
@derAlff  do you have any update on the progress of the adapter for using VRM? When are you planning to release?

**Comment by gidhap** [comment]
@derAlff I am back in the Victron Family :). So I am looking for the VRM adapter again.
Are you still working on this one? Txs for a quick update.
